frontend/model/state.js

```diff
@@ -623,6 +623,20 @@ sbp('okTurtles.events/on', CONTRACT_REGISTERED, (contract) => {
     store.registerModule(contract.name, {
       getters: omit(contract.getters, omitGetters[contract.name] || [])
     })
+
+    if (contract.name === 'gi.contracts/group') {
+      store.watch(
+        (state, getters) => getters.currentPaymentPeriod,
+        (newPeriod, oldPeriod) => {
+          if (!oldPeriod || !newPeriod) return
```

Comment:
Because this code feels a bit out-of-place and random, could you add a comment above this line explaining why this function/code is here, including the specific scenario it's addressing?